RELEASE_NOTES.md

```diff
@@ -1,5 +1,339 @@
+
+### HWRP Style Resources Now Require resource_name / provides
+
+Legacy HWRP style resources, those written as Ruby classes in the libraries directory of a cookbook, will now require either the `resource_name` or `provides` methods to be used to define the resource names. Previously Chef Infra Client would infer the desired resource name from the class, but this magic was problematic and has been removed.
+
+The `ChefDeprecations/ResourceWithoutNameOrProvides` cop in Cookstyle 6.0 and later detects this deprecation.
+
+### build_essential GCC Updated on Solaris
+
+On Solaris systems we no longer constrain the version of gcc to 4.8.2 in the build_essential resource to allow for gcc 5 installations.
+
+### s390x Packaging
+
+As outlined in our blog post at <https://blog.chef.io/chef-infra-end-of-life-announcement-for-linux-client-on-ibm-s390x-architecture/>, we will no longer be producing s390x platform packages for Chef Infra Client.
+
+### filesystem2 Node Data Replaces filesystem on FreeBSD / AIX / Solaris
+
+In Chef Infra Client 14 we introduced a modernized filesystem layout of Ohai data on FreeBSD, AIX, and Solaris at `node['fileystem2']`. With the release of 16.0 we're now replacing the existing data at `node['filesystem']` with this updated filesystem data. This data has a standardized format that matches Linux, macOS, and Windows data to make it eaier to write cross platform cookbooks. In a future release of Chef Infra Client we'll remove the `node['filesystem2']` as we complete this migration.
+
+### required: true on Properties Now Behaves As Expected
+
+The behavior of `required: true` has been changed to better align with th expected behavior. Previously if you set a property `required: true` on a custom resource property and did not explicitly reference the property in an action then Chef Infra Client would not raise an exception. This meant many users would add their own validation to raise for resources they wanted to ensure were always set. `required: true` will now properly raise if a property has not been set.
+
+We've also expanded the `required` field to allow you to specify actions where individual properties are required. This is especially useful when `:create` actions require certain properties that may not be required for a `:remove` type property.
+
+Example required field defining specific actions:
+
+```ruby
+property :password, String, required: [:create]
+
+action :create do
+  # code to create something
+end
+
+action :remove do
+  # code to remove it that doesn't need a password
+end
+```
+
+### Removal of Legacy metadata.rb depends Version Constraints
+
+Support for the `<<` and `>>` version constraints in metadata.rb has been removed. This was an undocumented feature from the Chef 0.10 era, which which is not used in any cookbooks on the Supermarket. We are mentioning it since it is technically a breaking change, but it unlikely that this change will be impacting.
```

Comment:
```suggestion
Support for the `<<` and `>>` version constraints in metadata.rb has been removed. This was an undocumented feature from the Chef 0.10 era, which is not used in any cookbooks on the Supermarket. We are mentioning it since it is technically a breaking change, but it unlikely that this change will be impacting.
```